DependencyInjection/Compiler/AddProcessorsPass.php

```diff
@@ -29,40 +29,65 @@ public function process(ContainerBuilder $container)
             return;
         }
 
+        $processors = [];
+
         foreach ($container->findTaggedServiceIds('monolog.processor') as $id => $tags) {
             foreach ($tags as $tag) {
-                if (!empty($tag['channel']) && !empty($tag['handler'])) {
-                    throw new \InvalidArgumentException(sprintf('you cannot specify both the "handler" and "channel" attributes for the "monolog.processor" tag on service "%s"', $id));
+                if (!isset($tag['priority'])) {
+                    $tag['priority'] = 0;
                 }
 
-                if (!empty($tag['handler'])) {
-                    $definition = $container->findDefinition(sprintf('monolog.handler.%s', $tag['handler']));
-                    $parentDef = $definition;
-                    while (!$parentDef->getClass() && $parentDef instanceof ChildDefinition) {
-                        $parentDef = $container->findDefinition($parentDef->getParent());
-                    }
-                    $class = $container->getParameterBag()->resolveValue($parentDef->getClass());
-                    if (!method_exists($class, 'pushProcessor')) {
-                        throw new \InvalidArgumentException(sprintf('The "%s" handler does not accept processors', $tag['handler']));
-                    }
-                } elseif (!empty($tag['channel'])) {
-                    if ('app' === $tag['channel']) {
-                        $definition = $container->getDefinition('monolog.logger');
-                    } else {
-                        $definition = $container->getDefinition(sprintf('monolog.logger.%s', $tag['channel']));
-                    }
-                } else {
-                    $definition = $container->getDefinition('monolog.logger_prototype');
-                }
+                $processors[] = [
+                    'id' => $id,
+                    'tag' => $tag,
+                ];
+            }
+        }
+
+        // Sort by priority so that higher-prio processors are added last.
+        // The effect is the monolog will call the higher-prio processors first
+        usort(
+            $processors,
+            function (array $left, array $right) {
+                return $left['tag']['priority'] <=> $right['tag']['priority'];
+            }
```

Comment:
Can the [`PriorityTaggedServiceTrait`](https://github.com/symfony/dependency-injection/blob/13b6032f8aeb4ff14bacd155121bcde9fde89721/Compiler/PriorityTaggedServiceTrait.php) be used here?

Reply:
We could still use the trait to get the sorted services and reverse the list before adding method calls.

Reply:
`\Symfony\Component\DependencyInjection\Compiler\PriorityTaggedServiceTrait::findAndSortTaggedServices` returns an array of references, but we need tag attributes like `handler` and `channel `.

Could we resolve this comment and merge this PR?

Reply:
@sudo-plz @stof I finished the PR in #485.

Reply:
@sudo-plz @stof can this PR be closed in favour of @HypeMC's one? It just wastes time, as I came here from Google and had to read all the discussions just to find out about continuation in different PR 😅. Also, this is great feature I would really welcome, so fingers crossed for it getting merged 🤞.